### ISSUE TITLE
Update installation instructions in README

### ISSUE THREAD

**Issue by paxtonfitzpatrick** [mortonne/psifr#3]
re: openjournals/joss-reviews#2669

Hey @mortonne, loving the package so far! I've come across just a few things while playing around with it that I'll put here and in other issues.  

I think a few updates to the [installation section of the README](https://github.com/mortonne/psifr/#installation) would be helpful:

- [x] Add an instruction for `pip`-installing the package (i.e., `pip install psifr`), probably before the part about cloning and doing a local install, since this will probably be the preferred method for most users.  I think this would make sense to have since [the install instructions appear in the PyPI listing](https://pypi.org/project/psifr/), plus it would make the README consistent with the [install instructions in the docs](https://psifr.readthedocs.io/en/latest/usage/installation.html#installation)
- [x] For the local installation instructions, I'd suggest changing `python setup.py install` to `pip install .`. I'm not sure whether this is entirely essential for the JOSS submission (the package still installs and runs fine as-is), but using `setuptools` directly is likely to cause issues down the road (and is marked for deprecation in favor of using pip) for a few reasons:
  - `setup.py install` uses the deprecated `egg` protocol rather than [wheels](https://www.python.org/dev/peps/pep-0427/)
  - `pip uninstall`ing a package installed via `setup.py install` often doesn't fully uninstall it
  - `setup.py install`ing a non-release version (e.g., the cloned repository at a non-release commit) can install duplicate or pre-release/unstable versions of dependencies.

I'd also recommend switching to `pip`-installing for [your CI test builds](https://github.com/mortonne/psifr/blob/671da414cbbb32615312da839a2f186e5cdfaf42/.travis.yml#L6), as the third issue particularly affects `setup.py develop` and can cause builds to fail unexpectedly.

**Comment by mortonne** [comment]
Thanks for pointing this out. I've had users run into problems that may have been related to the uninstallation problem you mentioned.
* Local installation is mostly relevant for developers, so I've moved that from `README.md` to `.github/CONTRIBUTING.md`. I've updated the instructions for local installation there.
* I've updated the installation instructions to match the documentation page instructions.
* The CI build configuration has been updated to use `pip`.

I'm leaving this issue open until I make a new release on PyPI, which will update the installation directions there.
